hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphIndexTransaction.java

```diff
@@ -766,7 +809,11 @@ private ConditionQuery constructSearchQuery(ConditionQuery query,
                 if (key instanceof Id && indexFields.contains(key)) {
                     // This is an index field of search index
                     Id field = (Id) key;
-                    String propValue = elem.<String>getPropertyValue(field);
+                    HugeProperty property = elem.getProperty(field);
+                    Object value = property.value();
+                    // collection join
+                    String propValue = value instanceof Collection ?
+                                       StringUtils.join(value) : (String)value;
```

Comment:
I think collection index is just related to secondary index, right?

Reply:
aso support search query 
 `elem.<String>getPropertyValue(field) `will cause exception